content/docs/atom-guide/dev-guide/shaders/azsl/_index.md

```diff
@@ -483,3 +483,64 @@ struct MyStruct
      [[pad_to(64)]] // This guarantees the sizeof(MyStruct) to be 64.
 };
 ```
+
+## class method, late declaration member access
+
+This scenario is a limitation of the reference tracking system internal to AZSLc: the seenat [more documentation](https://github.com/o3de/o3de-azslc/wiki/Features#seenats).
+  
+```cpp
+class C
+{
+	void f()
+	{
+		a;
+	}
+	int a;
+};
+```
+with CLI switch `--dumpsym` will result in:
+```yaml
+Symbol '/C/a':
+  kind: Variable
+  references:
+  line: 7
+  type:
+    core: {name: "?int", validity: found, tclass: Scalar, underlying_scalar: int}
+    generic: <NA>
+  storage: 
+  array dim: ""
+  has sampler state: no
+```
+The field `references` is empty, which means AZSLc didn't detect that `a` line 5, is the `a` delcared line 7.
+Let's look at the swapping of declaration sites of `f` and `a`:
+```cpp
+class C
+{
+  int a;
+	void f()
+	{
+		a;
+	}
+};
+```
+results in:
+```yaml
+Symbol '/C/a':
+  kind: Variable
+  references:
+    - {line: 6, col: 3}
+  line: 3
+  type:
+    core: {name: "?int", validity: found, tclass: Scalar, underlying_scalar: int}
+    generic: <NA>
+  storage: 
+  array dim: ""
+  has sampler state: no
+```
+Now it does see the access, at line 6, column 3 (after 2 tabs).
+
+The problem happens because AZSL does not parse in 2 sweeps, it has only one pass. f()'s content parsing happens before a's declaration, so 'a' apparition in f() is an internal unresolved reference. You can see it, using `--W3` warning level. The compiler will output:
+  
+> source.azsl(5,3) : warning: undeclared sub-symbol in idexpression: a
```

Comment:
Suggest putting this is a shell code block.

````suggestion
```shell
source.azsl(5,3) : warning: undeclared sub-symbol in idexpression: a
```
````